source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -328,9 +351,17 @@ namespace PlayFab
     {
         try
         {
-            // batch wasn't sent out due to an error, find it in the batch tracking map using customData pointer as a key
-            const auto foundBatchIterator = this->batchesInFlight.find(customData);
-            if (foundBatchIterator == this->batchesInFlight.end())
+            bool batchMissingFromFlight = false;
+            std::unordered_map<void*, std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>>>::iterator foundBatchIterator;
+
+            { // LOCK batchesInFlight mutex
+                std::unique_lock<std::mutex> lock(inFlightMutex);
+                // batch wasn't sent out due to an error, find it in the batch tracking map using customData pointer as a key
+                foundBatchIterator = this->batchesInFlight.find(customData);
+                batchMissingFromFlight = foundBatchIterator == this->batchesInFlight.end();
+            }
+            
+            if (batchMissingFromFlight)
```

Comment:
same comment about not access/writes to this->batchesInFlight over two locked blocks. Collapse into one.